server/routes/money.js

```diff
@@ -38,6 +77,65 @@ const createMoneyRouter = ({ hubContentService, offenderService }) => {
     }
   });
 
+  router.get('/transactions', async (req, res, next) => {
+    try {
+      const config = {
+        content: false,
+        header: false,
+        postscript: true,
+        detailsType: 'small',
+        returnUrl: req.originalUrl,
+      };
+
+      const { user } = req;
+
+      const data = {};
+
+      if (user) {
+        const accountTypes = ['spends', 'private', 'savings'];
+        const accountType = accountTypes.includes(req.query.accountType)
+          ? req.query.accountType
+          : accountTypes[0];
+
+        const toDate = new Date();
+        const fromDate = subDays(toDate, 30);
+
+        const prisonerInformation = await prisonerInformationService.getTransactionInformationFor(
+          user,
+          getAccountCodeFor(accountType),
+          fromDate,
+          toDate,
+        );
+
+        if (prisonerInformation) {
+          data.transactions = prisonerInformation.transactions.error
+            ? prisonerInformation.transactions
+            : prisonerInformation.transactions.map(formatTransaction);
+          data.balance = prisonerInformation.balances.error
+            ? prisonerInformation.balances
+            : {
+                amount: formatBalanceFor(
+                  accountType,
+                  prisonerInformation.balances,
+                ),
+              };
+        }
+
+        data.selected = accountType;
+        data.accountTypes = accountTypes;
+        config.userName = user.getFullName();
+      }
+
+      return res.render('pages/transactions', {
+        title: 'Your transactions',
+        config,
+        data,
+      });
+    } catch (exp) {
+      return next(exp);
+    }
```

Comment:
meh. We tend to wrap routes in async middleware so we don't have to remember to wrap all of our functions in try/catch... of course we then have to remember to adding the async middleware [everywhere](https://github.com/ministryofjustice/use-of-force/blob/main/server/routes/creatingReports/index.ts#L24)!

Reply:
Love the consistency/readability the wrapper functions adds ❤️ Carding this up as technical debt, I've bookmarked that link 👍